Code/GraphMol/ROMol.h

```diff
@@ -142,13 +143,14 @@ struct CXXBondIterator {
     Bond *current;
 
     CXXBondIter(Graph *graph, typename Graph::edge_iterator pos)
-        : graph(graph),
-          pos(pos),
-          current(boost::num_vertices(*graph) ? (*graph)[*pos] : 0) {}
+        : graph(graph), pos(pos), current(nullptr) {}
 
-    Edge &operator*() { return current; }
+    Edge &operator*() {
+      current = (*graph)[*pos];
```

Comment:
It seems like we don't need current anymore.

Reply:
That's what I thought initially, but you need it in order to be able to use references in the range for loops - `for(auto &at: mol.atoms())`.

Reply:
Right!  I'm doing some sanity checks here just trying to figure out when the end was being dereferences as I didn't catch it initially with valgrind.

Reply:
Valgrind only showed it for me when I built in debug mode. I noticed it first in a debug build on windows, where the dereference led directly to a crash.
The problem itself was happening here:
https://github.com/rdkit/rdkit/blob/modern_cxx/Code/GraphMol/ROMol.h#L115
when this call is made:
https://github.com/rdkit/rdkit/blob/modern_cxx/Code/GraphMol/ROMol.h#L131